GUI/src/main/kotlin/frameNavigation/FrameNavigation.kt

```diff
@@ -122,34 +124,56 @@ class FrameNavigation(state: MutableState<AppState>) : FrameNavigationInterface
         val diffFrame = ((diffSequence.size - 1).toDouble() * percentage).roundToInt()
 
         // check if the frame is already displayed
-        if (diffFrame == currentIndex.value) {
+        if (diffFrame == currentIndex) {
             return
         }
 
         // jump to the frame
-        jumpToFrame(diffFrame)
+        currentIndex = diffFrame
+        jumpToFrame()
     }
 
     /**
      * Jump to a concretely specified frame.
      * @param index [Int] containing the index of the frame to jump to.
      * @return [Unit]
      */
-    override fun jumpToFrame(index: Int) {
-        // check bounds
-        val boundedIndex = index.coerceIn(0, diffSequence.size - 1)
-        currentIndex.value = boundedIndex
-        currentRelativePosition.value = boundedIndex.toDouble() / (diffSequence.size - 1).toDouble()
-
-        // jump to the frame in each video by mapping the frame using the generated sequences
-        video1Grabber.setVideoFrameNumber(video1Frames[boundedIndex])
-        video2Grabber.setVideoFrameNumber(video2Frames[boundedIndex])
-        grabberDiff.setVideoFrameNumber(boundedIndex)
-
-        // update the bitmaps
-        video1Bitmap.value = getBitmap(video1Grabber)
-        video2Bitmap.value = getBitmap(video2Grabber)
-        diffBitmap.value = getBitmap(grabberDiff)
+    override fun jumpToFrame() {
+        var coercedIndex = currentIndex.coerceIn(0, diffSequence.size - 1)
+        currentRelativePosition.value = coercedIndex.toDouble() / (diffSequence.size - 1).toDouble()
+        if (lock) {
+            return
+        }
+        lock = true
+        var goal = -1
+        scope.launch(Dispatchers.IO) {
+            var b1: ImageBitmap? = null
+            var b2: ImageBitmap? = null
+            var b3: ImageBitmap? = null
+
+            while (goal != coercedIndex) {
+                goal = coercedIndex
+
+                // update the bitmaps
+                video1Grabber.setVideoFrameNumber(video1Frames[goal])
+                b1 = getBitmap(video1Grabber)
+
+                video2Grabber.setVideoFrameNumber(video2Frames[goal])
+                b2 = getBitmap(video2Grabber)
+
+                grabberDiff.setVideoFrameNumber(goal)
+                b3 = getBitmap(grabberDiff)
+
+                coercedIndex = currentIndex.coerceIn(0, diffSequence.size - 1)
```

Comment:
Please explain here, why you set the index again (I know, it's because of the coroutines, but we need some more documentation here)